### ISSUE TITLE
Automatic full-measure rests that match the time signature

### ISSUE THREAD

**Issue by rpatters1** [musescore/MuseScore#24615]
### Your idea

I am absolutely gobsmacked (in an excellent way) that MuseScore automatically switches to using double-whole rests for default full-measure rests in time signatures of 4/2 and larger. Kudos, kudos, kudos. Even better, it correctly still uses a whole rest in single-measure multimeasure rests using the old symbols. The Breitkopf editors from 1890 are resting peacefully now. Neither Finale nor (it seems) Dorico does this. I don't know about Sibelius, but I would bet not.

What appears to be missing is the ability to automatically fill empty measures with rests according to the time signature. My proposal is to add this as an option to the time signature definition.

### Problem to be solved

In complex meter changing situations, especially when using uneven meters like 5/8 or 7/8, it is extremely useful to musicians to show the beat pattern in empty measures.

### Prior art

I do not know of a current music program that does it automatically, but it is not uncommon in certain styles of the 20th century and later. Elaine Gould explicitly condones it on p. 160 of *Behind Bars*.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by wizofaus** [comment]
You might be gobsmacked it does that, I'm gobsmacked you have no choice over the matter, and raised this a long while back: https://github.com/musescore/MuseScore/issues/22129.
Breve-style rests make no sense for time signatures like 11/4 (or 17/8 etc. etc.).  Personally I never want to see them, regardless of what Gould recommends.

**Comment by rpatters1** [comment]
I'm all for options. Indeed, I am also gobsmacked there doesn't seem to be a simple option to turn off full-measure rests entirely.

As to the example of the filling with rests, it is certainly a situation that I have wished for at times. I know I have seen it, but I am coming up blank on an example. The problem with using "real" rests is that they prevent multimeasure rests.

**Comment by MarcSabatella** [comment]
Seems to me the best place for options controlling rest behavior is in time signature properties.  That way you can have a 4/2 section set to use breve rests, a 7/8 section set to use duration rests, and a 6/4 section set to use regular whole rests, all in the same score.

Along with that there would need to be a way to control the distribution of those duration rests, and a way to set beam behavior of the rests.

**Comment by rpatters1** [comment]
>you can have a 4/2 section set to use breve rests, a 7/8 section set to use duration rests, and a 6/4 section set to use regular whole rests, all in the same score.

I agree. As long as we're about it, I would say we need an option for no empty-measure rests. (I've seen this in Messiaen scores, for example: not cutaway but no rests in empty measures.)

FWIW: 6/4 currently (and correctly) gets whole rests, not double-whole. (Nitpick, sorry.)

**Comment by rpatters1** [comment]
One concern I have for putting it on the time signature is if there a lot of time signature changes, it could become tedious to have to choose the option for each time sig. There might be mitigations possible.

**Comment by MarcSabatella** [comment]
Just as with custom beaming, this is already simple enough to handle - just add your customized time signature back to the palette (Ctrl+Shift+drag) for easy reuse.

**Comment by rpatters1** [comment]
This gets into an issue with the palette: telling which version of 7/8 (e.g. 3+2+2 or 2+2+3 or 2+3+2) or 5/8 it is. But perhaps that is a discussion for another issue. (And also maybe I missed something being a newbie.)

**Comment by MarcSabatella** [comment]
Use the cell properties to set the name/description, which becomes the tooltip.

**Comment by wizofaus** [comment]
Agree being able to override at the time signature level would be the most flexible, and given time signatures where you want to control such behaviour are relatively unusual it's probably adequate - providing palette customisations are preserved reliably, even when running multiple instances.

**Comment by rpatters1** [comment]
I'm thinking the most flexible option would be a score level setting and a per-timesig setting with `Auto` as an option.

The score setting could be `Display rests in empty measures:`

The Score options would be:

```
Whole and double-whole for 4/2 and higher
Whole
None
```

The time-sig settings could be:

```
Auto
Whole and double-whole for 4/2 and higher
Whole
None
Time signature
```

I left "Time signature" off the score list, though we could add it. But I'm thinking it makes no sense as a score-level setting. Rather, I see it truly being a per-timesig override. If others think it should be on the score list, I'm cool with that.

**Comment by rpatters1** [comment]
I finally was reminded of an example of a score from a mainstream publisher where full-measure rests match the time sig.

<img width="607" height="502" alt="Image" src="https://github.com/user-attachments/assets/866f41b1-5aff-4e1b-8669-e48e5688cde7" />

This is from the Boosey and Hawkes edition of John Adams' *Chamber Symphony*. It appears that the style they are using is that typical time signatures in quarters use whole rests, no matter their duration, but odd valued or small time sigs in 8ths fill with rests matching the time sig.

This edition is [online](https://www.boosey.com/cr/perusals/score?id=981) and available to view at no cost. (You have to sign up, but it costs nothing.)
